### ISSUE TITLE
Root .haccess leads to non-existent index.php

### ISSUE THREAD

**Issue by dantaeusb** [magento/magento2#32409]
I'm sorry if I'm wrong, but I'm extremely confused about the installation process. I followed steps included in official guides, with basic configuration based on Docker and WSL2. I tried to follow every step carefully several times cause still don't believe that's not my fault. However, looking at the older versions of source code, I'm very confused that there are no adjustments in docs and .htaccess about removing index.php in the root directory ([commit 640cad53009b291334234ccd61ab79f256b43da2](https://github.com/magento/magento2/commit/640cad53009b291334234ccd61ab79f256b43da2#diff-7413d6453f901e939bbd840c8f0d1c7b20c2ca0e7f71741e4e07c6cf036f16c0)). There were some links to #356 but it doesn't seem to cover the actual state.

**TL;DR: index.php seem no longer exist in root folder, and both .htaccess and official docs assuming it does;** the [Apache configuration page](https://devdocs.magento.com/guides/v2.4/install-gde/prereq/apache.html) does not cover this change however there seems to be an update that covers [elasticsearch](https://devdocs.magento.com/guides/v2.4/install-gde/prereq/es-config-apache.html) installation info.

### Preconditions (*)
1. Try to install Magento 2.4.2
2. Get 403's or file not found after installation and creating vhosts to repo root.

### Steps to reproduce (*)
1. Follow official documentation at https://devdocs.magento.com/guides/v2.4/install-gde/composer.html

### Expected result (*)
Installation complete and site works assuming `.htaccess` in root actually forwards requests to index.php in root.

### Actual result (*)
Line 7 of `.htaccess` states `DirectoryIndex index.php` which is basically non-existent in version 2.4. Adjusting default configuration to `pub` directory doesn't work and seems like requires some additional changes, which is not covered in guides mentioned above.

### Additional info

Apache image used: https://github.com/bitnami/bitnami-docker-apache
PHP-FPM image used: https://github.com/bitnami/bitnami-docker-php-fpm + needed extensions for Magento 2
This page partially covers what seems to be necessary now: https://devdocs.magento.com/guides/v2.4/install-gde/tutorials/change-docroot-to-pub.html

**Comment by dantaeusb** [comment]
My configuration is incorrect, I was trying to set up Apache + php-fpm because I haven't seen mod_php in requirements neither on [this](https://devdocs.magento.com/guides/v2.4/install-gde/prereq/apache.html) nor [this](https://devdocs.magento.com/guides/v2.4/install-gde/prereq/php-settings.html) page, but it seems to be assumed "by default" for Apache installations.

The Issue could be irrelevant, though I still wonder about the purpose of `.htaccess` in root and whether it works if installed by following guide.

**Comment by mrtuvn** [comment]
You should update description for what version of magento that you got issue
Since 2.4.2 magento changed root folder to pub (index.php file in there) No longer file index.php in root like before (2.1, 2.2, 2.3)
If you have only one site magento at time run. Everything start at pub folder

If you see file .htaccess that changed

`RewriteEngine on
RewriteCond %{REQUEST_URI} !^/pub/
RewriteCond %{REQUEST_URI} !^/setup/
RewriteCond %{REQUEST_URI} !^/update/
RewriteCond %{REQUEST_URI} !^/dev/
RewriteRule .* /pub/$0 [L]
DirectoryIndex index.php`

**Comment by dantaeusb** [comment]
Hi Tu,

Sure thing, I was talking about Magento 2.4.2. Yes, practically I understood that "document_root_is_pub" is now a default behavior. From my point of view, the problem is:

a) This change is not covered by official guides at all;
b) It's meaningless and confusing because it's always overridden by similar rules `pub/.htaccess`. I was thinking that I have deployment issues because I was desperately looking for index.php in root folder, which this file refers to.

**Comment by mrtuvn** [comment]
Yes must be someone need to update docs soon. This critical may affected to a lot store when upgrade to new version
You should try to point to index.php file in pub instead. Maybe that will help
